src/executables/master.c

```diff
@@ -740,6 +792,7 @@ css_send_to_existing_server (CSS_CONN_ENTRY * conn, unsigned short rid, CSS_SERV
 	      css_send_data (conn, rid, (char *) &buffer, sizeof (int));
 	      buffer = htonl (temp->port_id);
 	      css_send_data (conn, rid, (char *) &buffer, sizeof (int));
+	      conn_established_flag = true;
```

Comment:
The client request is rejected here too. See that it falls through to `css_reject_client_request`.